### DISCUSSION TITLE
Using solara.Link with query parameters

### DISCUSSION THREAD

**Discussion by HugoP** [widgetti/solara#682] [Q&A]
Hi,

I'm trying to link to another page and pass the page query parameters like so.

```python
with solara.Link("/anotherpage?my_parameter=my_data"):
    solara.Text(text="Go to page")                
```
The doc's suggest the following [Custom state storage](https://solara.dev/documentation/examples/general/custom_storage)

But the Link component does not have a on_click event to use this method.

Does Solara support query parameters?

Kind regards
Hugo

**Comment by maartenbreddels** [comment]
Hi,

this is certainly possible, the docs describe it here:
https://solara.dev/documentation/api/routing/use_router

I've put together an example at:
https://py.cafe/maartenbreddels/solara-query-params
(Note that pycafe does not yet reflect the search query in the address bar, so maybe you want to run this example locally)

See also https://github.com/widgetti/solara/pull/34

Does this help you?

Regards,

Maarten

**Comment by HugoP** [comment]
Hi,

Thank you, great stuff.

Regards,

Hugo
